### PR TITLE
Added optimizer for lm_alpha + lm_beta

### PR DESCRIPTION
**Pull request by kdavis-mozilla**

### CODE REVIEW

File: optimizer.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+
+def is_character_based():
+    character_based = False
+    if FLAGS.scorer_path:
+        scorer = Scorer(FLAGS.lm_alpha, FLAGS.lm_beta, FLAGS.scorer_path, Config.alphabet)
+        character_based = scorer.is_utf8_mode
+    return character_based
+
+CHARACTER_BASED = is_character_based()
```
Comment (rhamnett) [inline optimizer.py]: This needs to be moved into the main() function after the initialize so that flags are initialized

Comment (kdavis-mozilla) [reply optimizer.py]: Thanks. I'll move this around.

**Comment by rhamnett** [comment]
Good work :)

**Comment by piraka9011** [comment]
This is awesome, thanks! I was just going to go about implementing my own :)
I'd like to give this a shot on my own model. It doesn't look like I need to recompile DS for this, correct?

**Comment by kdavis-mozilla** [comment]
Thanks, and you don't need to re-compile.

**Comment by kdavis-mozilla** [comment]
@reuben Do you have any ideas? Maybe `lm_optimizer.py`?

**Comment by reuben** [comment]
Yeah I like `lm_optimizer.py`.

**Pull request merged**